core/state_transition.go

```diff
@@ -515,6 +515,7 @@ func (st *stateTransition) execute() (*ExecutionResult, error) {
 	if rules.IsPrague {
 		// After EIP-7623: Data-heavy transactions pay the floor gas.
 		if st.gasUsed() < floorDataGas {
+			gasRefund = 0
 			prev := st.gasRemaining
 			st.gasRemaining = st.initialGas - floorDataGas
```

Comment:
Not directly related to this PR, but I'm trying to figure out why we don't also subtract the base cost (21000) here.